src/main/java/uk/gov/pay/publicauth/resources/PublicAuthResource.java

```diff
@@ -189,13 +189,13 @@ private void validatePayloadHasFields(JsonNode payload, List<String> expectedFie
                 .filter(expectedKey -> !payload.has(expectedKey))
                 .collect(Collectors.toList());
         if (!missingFields.isEmpty()) {
-            throw new ValidationException("Missing fields: [" + Joiner.on(", ").join(missingFields) + "]");
+            throw new ValidationException(format("Missing fields: [%s]", missingFields));
```

Comment:
Not quite. [`toString()`](https://docs.oracle.com/javase/8/docs/api/java/util/AbstractCollection.html#toString--) will add the square brackets for you, so you want this:

```java
throw new ValidationException(format("Missing fields: %s", missingFields));
```

But I’d just use string concatenation for this:

```java
throw new ValidationException("Missing fields " + missingFields);
```

Reply:
I love it when other people are correct